CONTRIBUTING.md

```diff
@@ -124,3 +124,73 @@ what a commit is about.
 
 When you create a pull request, please
 [make your PR editable](https://github.com/blog/2247-improving-collaboration-with-forks).
+
+## Rebasing
+
+If other PRs have been merged during the time between your initial PR creation
+and final approval, it may be required that you rebase your changes against the
+latest `main` branch.
+
+There are potential pitfalls here if you follow the suggestions from `git`,
+which could leave your branch in an unrecoverable mess,
+and you having to start over with a new branch and new PR.
```

Comment:
I have found suggestions from git to always be pretty good at guessing the likely use-cases of my current operation. Where will they lead users astray?

Reply:
If you try `git push` after rebasing locally and resolving conflicts, it will suggest

```
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

If you follow the `git pull` hint, it will splatter your local branch with a duplicate set of commits, undoing the rebase and all the conlflict resolutions.  So the branch becomes a mess and you have to start over.

Maybe this is because I am doing it wrong.  But documentation on best practices is thin on the ground.